src/components/recommendation.tsx

```diff
@@ -0,0 +1,75 @@
+import { useQuery } from "@tanstack/react-query";
+
+import { mockFetch } from "@/lib/mock-fetch";
+import type { RecommendationResponse } from "@/types/api-types";
+import type { Chat } from "@/types/chat";
+
+import { PromochatorIcon } from "./chat";
+import { Supervisor } from "./supervisor";
+
+function useRecommendationQuery(
+  chat: Chat,
+  updateChat: (uuid: string, _chat: Partial<Chat>) => void,
+) {
+  return useQuery({
+    queryKey: ["recommendation", chat.uuid],
+    queryFn: async () => {
+      const response = await mockFetch("/api/recommend");
+      // const response = await fetch("/api/recommend", {
+      //   method: "POST",
+      //   body: JSON.stringify({
+      //     input: { question: chat.prompt, faculty: "" },
+      //   }),
+      // });
+
+      const data = (await response.json()) as RecommendationResponse;
+      updateChat(chat.uuid, { recommendation: data.recommendation });
+      return data;
+    },
```

Comment:
**suggestion:** I know this is commented out for now, but couldn't we use the `fetchData` function from `src/lib/api` in this case as well? This way we don't need to do an assertion, we can just pass `RecommendationResponse` as the generic
```suggestion
      const data = await fetchData<RecommendationResponse>("/api/recommend", {
        method: "POST",
        body: JSON.stringify({
          input: { question: chat.prompt, faculty: "" },
        }),
      });
      updateChat(chat.uuid, { recommendation: data.recommendation });
      return data;
```

Reply:
I think we can't because of how we set the API_URL env variable - in current implementation it's not available on client ([why](https://nextjs.org/docs/app/building-your-application/configuring/environment-variables#bundling-environment-variables-for-the-browser)) and the whole idea with creating api route was to hide it

Reply:
1. It's used in api route so don't really see the point
2.`fetchData` is configured to call only promochator backend api (not the best solution I guess xd)

Reply:
Ok, let's just stay with this then